canteenWeb/serializers.py

```diff
@@ -67,3 +68,34 @@ class MenuItemSerializer(serializers.ModelSerializer):
     class Meta:
         model = MenuItem
         fields = ("id", "name", "price", "is_available", "preparation_time", "options")
+
+
+class SignUpSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = (
+            "first_name",
+            "last_name",
+            "email",
+            "is_staff",
+            "username",
+            "password",
+        )
+        extra_kwargs = {"password": {"write_only": True}}
+
+    def create(self, validated_data):
+        user = User(
+            first_name=validated_data["first_name"],
+            last_name=validated_data["last_name"],
+            email=validated_data["email"],
+            is_staff=validated_data["is_staff"],
+            username=validated_data["username"],
+        )
+        user.set_password(validated_data["password"])
+        user.save()
+        return user
+
```

Comment:
We need Profile serializers as well for student and teacher profiles, but I guess that'll be a different issue. @preet4737 You can work on that.